### PR TITLE
[ICDS] - Added is_test property to the ucr and aggregations.

### PR DESCRIPTION
**Pull request by lwyszomi**
Hi @emord @calellowitz 

I added is test property for the UCR awc locations and also I extended two first aggregation script by these properties.

This PR is already big so updates for the other aggregations I will send in separate PRs.

Please let me know if you have any questions.

Regards,
Lukasz

### CODE REVIEW

**Comment by lwyszomi** [comment]
Hi @calellowitz Could you kick off the build I think failure is not related

**Comment by calellowitz** [comment]
restarted the part that failed

**Comment by calellowitz** [comment]
@lwyszomi sorry for the delay on this. generally speaking it looks good, although i think emord's comments about transaction handling on the ucr database holds here. `transaction.atomic()` needs to be `transaction.atomic(using=<ucr_db>)` you can examples of that in the other aggregations now, or in the tasks file. once the conflict is resolved, I will do a final pass

**Comment by lwyszomi** [comment]
@calellowitz I resolved conflicts. After merge PR is smaller so I think we don't need to split it.

**Comment by lwyszomi** [comment]
Could you kick off the build I think failure is not related

File: custom/icds_reports/utils/aggregation_helpers/agg_awc.py
Comment (snopoke) [inline custom/icds_reports/utils/aggregation_helpers/agg_awc.py]: `aggregation_level` here will never be > 4 so these last 2 will always be 0. Is that correct?

**Comment by snopoke** [comment]
High level question, do we even want the test location data in the DB? Surely it will just get filtered out and never displayed on the UI?

File: custom/icds_reports/utils/aggregation_helpers/agg_awc.py
Comment (snopoke) [inline custom/icds_reports/utils/aggregation_helpers/agg_awc.py]: Grouping by `is_test` seems a bit weird to me. I'd have thought you'd want `max(supervisor_is_test)` which will tell you if  the supervisor was ever 'test' in the time period.

**Comment by lwyszomi** [comment]
@snopoke We need data for test location because when ICDS start training for new location then it is marked as test location to don't show data together with the locations which already started collecting real data. Currently, we collect data for test location and we can show it on the dashboard adding a flag to the URL, but we found an issue that we can exclude test data for specific level for example supervisor. but when we aggregate data for the block we have data from all location tests and real. More details in case: http://manage.dimagi.com/default.asp?282106#1541838

**Comment by snopoke** [comment]
@lwyszomi thanks for the explanation. I'll try get through the rest of the review today.

File: custom/icds_reports/utils/aggregation_helpers/agg_awc.py
Comment (lwyszomi) [reply custom/icds_reports/utils/aggregation_helpers/agg_awc.py]: @snopoke True, but we aggregate data for a specific month and calculate data by location below, so for example when we collecting data for supervisor we take data for the specific month for all children. I think we can't have a possibility that one child will have ```is_test``` for parent ```True``` and one ```False```

File: custom/icds_reports/utils/aggregation_helpers/agg_awc.py
Comment (lwyszomi) [reply custom/icds_reports/utils/aggregation_helpers/agg_awc.py]: :+1:

**Comment by lwyszomi** [comment]
@calellowitz @calellowitz Updated

**Comment by calellowitz** [comment]
looks like a bunch of new merge conflicts

**Comment by lwyszomi** [comment]
@calellowitz I resolved conflicts

File: custom/icds_reports/utils/aggregation_helpers/agg_awc_weekly.py
Comment (calellowitz) [inline custom/icds_reports/utils/aggregation_helpers/agg_awc_weekly.py]: does `ut` need to be after the `FROM` clause not the `WHERE` clause

File: custom/icds_reports/utils/aggregation_helpers/agg_awc_weekly.py
Comment (calellowitz) [inline custom/icds_reports/utils/aggregation_helpers/agg_awc_weekly.py]: Where does `agg_awc` come from?

File: custom/icds_reports/utils/aggregation_helpers/agg_awc_weekly.py
Comment (calellowitz) [inline custom/icds_reports/utils/aggregation_helpers/agg_awc_weekly.py]: same questions about ut and agg_awc

**Review by calellowitz — COMMENTED** [review]
i left a couple sql questions that i think you did not introduce but look like bugs, happy to merge if you prefer or we can fix them now. i also know @Rohit25negi was doing some simultaneous work on `agg_awc_weekly` so it may even be easier to just drop those changes and let him move it into python in his pr

**Comment by lwyszomi** [comment]
@calellowitz I restored weekly aggregation to the original implementation.

File: custom/icds_reports/migrations/sql_templates/database_functions/update_location_table.sql
Comment (snopoke) [inline custom/icds_reports/migrations/sql_templates/database_functions/update_location_table.sql]: I assume there'll be a follow up PR to delete these functions

Comment (lwyszomi) [reply custom/icds_reports/migrations/sql_templates/database_functions/update_location_table.sql]: Yes, I will send next PR when this will be merged.

File: custom/icds_reports/utils/aggregation_helpers/awc_location.py
Comment (lwyszomi) [reply custom/icds_reports/utils/aggregation_helpers/awc_location.py]: int, updated

File: custom/icds_reports/utils/aggregation_helpers/agg_awc.py
Comment (lwyszomi) [reply custom/icds_reports/utils/aggregation_helpers/agg_awc.py]: Now we update test properties when it is null or if we have a difference between an updated table and awc_location.

File: custom/icds_reports/models/aggregate.py
Comment (lwyszomi) [reply custom/icds_reports/models/aggregate.py]: moved to the tasks.py file

File: custom/icds_reports/tasks.py
Comment (lwyszomi) [reply custom/icds_reports/tasks.py]: mistake, I forgot to remove after tests

**Comment by lwyszomi** [comment]
@calellowitz @snopoke Updated

**Review by calellowitz — APPROVED** [review]

**Comment by calellowitz** [comment]
looks good to me. will probably merge when tests pass unless @snopoke objects. do you plan on changing the views to use these new properties to exclude test locations more fully?

**Review by snopoke — APPROVED** [review]

**Pull request merged**
